administration/tests/test_admin_profiles.py

```diff
@@ -0,0 +1,330 @@
+        ]
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual(data, response.json())
+
+    def test_get_profile_id_authenticated(self):
+        self.client.force_authenticate(self.user)
+        response = self.client.get(
+            path=f"/api/admin/profiles/{self.profile.id}/"
+        )
+        data = {
+            "name": "Test person",
+            "is_registered": True,
+            "is_startup": True,
+            "categories": [],
+            "activities": [],
+            "person": {
+                "name": "Test person 1",
+                "surname": "Test person 1 surname",
+                "email": "test1@test.com",
+                "is_active": True,
+                "is_staff": True,
+                "is_superuser": False,
+                "company_name": True,
+            },
+            "person_position": "Test person position",
+            "official_name": "Test official name",
+            "regions": [],
+            "common_info": "test common info",
+            "phone": "380112909099",
+            "edrpou": "10000000",
+            "founded": 2022,
+            "service_info": "test service info",
+            "product_info": "test product info",
+            "address": "Test Country, Test City, St. Test, 1",
+            "startup_idea": "Test startup idea",
+            "banner_image": "http://testserver/media/Test_banner_image",
+            "is_deleted": False,
+        }
+        self.assertEqual(data, response.json())
+
+
+class TestDeleteProfile(APITestCase):
+    def setUp(self):
+        self.profile = AdminProfileFactory()
+        self.user = AdminUserFactory()
+
+    def test_delete_Profile(self):
+        self.client.force_authenticate(self.user)
+        response = self.client.delete(
+            path=f"/api/admin/profiles/{self.profile.id}/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_delete_profile_not_authorized(self):
+        response = self.client.delete(
+            path=f"/api/admin/profiles/{self.profile.id}/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestPutProfile(APITestCase):
+    def setUp(self):
+        self.profile = AdminProfileFactory()
+        self.user = AdminUserFactory()
+
+    def test_put_profile(self):
+        self.client.force_authenticate(self.user)
+        data = {
+            "name": "Test string",
+            "is_registered": True,
+            "is_startup": True,
+            "person_position": "Test string",
+            "official_name": "Test string",
+            "common_info": "Test string",
+            "phone": 123456789012,
+            "edrpou": 12345678,
+            "founded": 2024,
+            "service_info": "Test string",
+            "product_info": "Test string",
+            "address": "Test string",
+            "startup_idea": "Test string",
+            "is_deleted": True,
```

Comment:
I assume, that we should have separate tests for updating fields and for marking the company as deleted.

Reply:
There are no requirements for the administrative panel. A comprehensive test has been made